Feature/infrastructure database setup

AlexBud369 opened the pull request:
Initial database in infrastructure layer

- **Added NuGet packages** to `Infrastructure`:
  - `Microsoft.EntityFrameworkCore.SqlServer` (v9.0.0)
  - `Microsoft.EntityFrameworkCore.Design` (v9.0.0)
  - `Microsoft.AspNetCore.Identity.EntityFrameworkCore` (v9.0.0)

- **Created `AppDbContext.cs`**:
  - Inherits `IdentityDbContext<ApplicationUser, ApplicationRole, Guid>`
  - Configures `Book` and `Tag` entities with relationships
  - Adds model constraints (title, author, tag length limits)

- **Updated `appsettings.json`**:
  - Added SQL Server connection string
  - Included base JWT configuration

AlexBud369 committed:
feat: add EF Core and Identity NuGet packages to Infrastructure
- Microsoft.EntityFrameworkCore.SqlServer: Enables SQL Server database provider for EF Core
- Microsoft.EntityFrameworkCore.Design: Supports EF Core migrations and CLI tools
- Microsoft.AspNetCore.Identity.EntityFrameworkCore: Integrates ASP.NET Core Identity with EF Core for user and role management

AlexBud369 committed:
feat: add AppDbContext for EF Core database configuration
- Implements IdentityDbContext for user and role management
- Defines many-to-many relationship between Book and Tag
- Defines one-to-many relationship between Book and ApplicationUser

AlexBud369 committed:
feat: add connection string and JWT settings
- Configures SQL Server connection for EF Core
- Adds JWT configuration for authentication

budnikme commented on src/Infrastructure/Data/AppDbContext.cs:
```diff
@@ -0,0 +1,44 @@
+        builder.Entity<Book>()
+            .HasMany(b => b.Tags)
+            .WithMany(t => t.Books)
+            .UsingEntity(j => j.ToTable("BookTags"));
```
Why does ToTable mapping is needed here?

budnikme commented on src/Infrastructure/Data/AppDbContext.cs:
```diff
@@ -0,0 +1,44 @@
+
+    protected override void OnModelCreating(ModelBuilder builder)
+    {
+        base.OnModelCreating(builder);
```
I guess it woould be worth to create migrations/initial migration as part of this PR

AlexBud369 replied:
I can't add migrations to this pull request because this branch doesn't have the domain entities required for migrations. This branch was created from develop, which didn't have any entities because I didn't merge this pull request [https://github.com/AlexBud369/BookSharingPlatform/pull/3](https://github.com/AlexBud369/BookSharingPlatform/pull/3) into it.

AlexBud369 committed:
feat: update AppDbContext
- Added DbSet<RefreshToken> and configured RefreshToken -> ApplicationUser relationship
- Removed ToTable('BookTags') to use default table name (per reviewer comment)
- Fixed TagName property in Tag configuration
- Added configurations for Book.CoverImageUrl, ApplicationUser.IsBlocked, and ApplicationUser.CreatedAt
- Replaced magic numbers with DomainConstants

AlexBud369 committed:
feat: Add Azure Blob Storage configuration to appsettings.json

AlexBud369 committed:
feat: update AppDbContext for book cover feature
- Added MaxLength constraint to Book.CoverImageUrl using DomainConstants.BookCover.UrlMaxLength
- Removed unused Azure using directive

AlexBud369 committed:
Merge branch 'develop' into feature/infrastructure-database-setup
